app/validators/workflow_execution_advanced_search_group_validator.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+# Validator for workflow execution advanced search groups
+# Validates search conditions including field names, operators, and values
+# Specific to WorkflowExecution attributes and metadata fields
+class WorkflowExecutionAdvancedSearchGroupValidator < ActiveModel::Validator
```

Comment:
This file is nearly identical to the `advanced_search_group_validator.rb`. May want to have a base validator with the identical methods within. I noticed just these two methods are different between the two files

```
def validate_date_and_numeric_field(condition)
def validate_key(condition)
```

Reply:
Updated in dd22704aa59ecdf2ef35dcaed4c13f4e8970b12c